platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/ConsensusTransactionImpl.java

```diff
@@ -16,60 +16,51 @@
 
 package com.swirlds.platform.system.transaction;
 
-import static com.swirlds.platform.system.events.ConsensusData.NO_CONSENSUS;
-
 import java.time.Instant;
 
 /**
  * A transaction that may or may not reach consensus.
  */
 public abstract non-sealed class ConsensusTransactionImpl implements ConsensusTransaction {
-
-    /**
-     * The consensus order of the event that contains this transaction, or -1 if consensus has not yet been reached.
-     * NOT serialized and not part of object equality or hash code
-     */
-    private long consensusOrder = NO_CONSENSUS;
-
     /**
      * The consensus timestamp of this transaction, or null if consensus has not yet been reached.
      * NOT serialized and not part of object equality or hash code
      */
     private Instant consensusTimestamp;
+    /** An optional metadata object set by the application */
+    private Object metadata;
```

Comment:
"Metadata" sounds generic, and so does the plain `Object` type. Above I've seen it's used for signatures only.

Should this field (and its getters) be named and typed accordingly? If not, then its javadoc could mention a few different use-cases to clarify why it's loosely typed.

Reply:
This is old code, just moved. The interface declaring `getMetadata` and `setMetadata` explains what its used for. Our test apps use it for signatures, Hedera sets a different type.

Reply:
Got it. Makes sense. Thanks for the pointer.